java-checks-test-sources/default/src/main/java/checks/PrintfMisuseCheck.java

```diff
@@ -254,6 +254,10 @@ void foo(Calendar c) throws IOException {
     logger4.log(level, "message " + "...");
     logger4.log(level, "message " + param1, new Exception()); // Noncompliant {{Lambda should be used to defer string concatenation.}}
 
+    logger4.log(level, "Can't load library \"{0}\"!", "foo"); // Noncompliant {{Single quote "'" must be escaped.}}
```

Comment:
The message says "Single quote "'" must be escaped.". How is it supposed to be escaped? `Can\'t` or `Can''t`? 
- `Can\'t`: it still raises an issue. If this is the supposed escape, then the rule must include an exception for it.
- `Can''t`: I'm not sure it produces the expected output "Can't", we should make this sure by checking the docs or test it ourselves if not documented explicitly.